crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -135,13 +120,28 @@ impl Extension for QueryLimitsChecker {
         variables: &Variables,
         next: NextParseQuery<'_>,
     ) -> ServerResult<ExecutableDocument> {
+        let query_id = ctx.data_unchecked::<Uuid>().to_string();
+        let session_id = ctx.data_unchecked::<SocketAddr>().to_string();
+        let metrics = ctx.data_unchecked::<Metrics>();
```

Comment:
Same comment here on ascribing the type on the variable and avoid the `to_string()`.

```suggestion
        let query_id: Uuid = ctx.data_unchecked();
        let session_id: SocketAddr = ctx.data_unchecked();
        let metrics: Metrics = ctx.data_unchecked();
```

Reply:
Same as before, it will not work for the `query_id` and `session_id` variables.